### ISSUE TITLE
pdf not converted

### ISSUE THREAD

**Issue by ContaTP** [jupyter/nbconvert#255]
#I am using Python 2.7 and Jupyter 4.0.6
When I try to convert my notebook to pdf, get the error below:
```
Traceback (most recent call last):
  File "d:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "d:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "D:\Python27\Scripts\jupyter-nbconvert.exe\__main__.py", line 9, in <module>
  File "d:\python27\lib\site-packages\jupyter_core\application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "d:\python27\lib\site-packages\traitlets\config\application.py", line 589, in launch_instance
    app.start()
  File "d:\python27\lib\site-packages\nbconvert\nbconvertapp.py", line 289, in start
    self.convert_notebooks()
  File "d:\python27\lib\site-packages\nbconvert\nbconvertapp.py", line 412, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "d:\python27\lib\site-packages\nbconvert\nbconvertapp.py", line 383, in convert_single_notebook
    output, resources = self.export_single_notebook(notebook_filename, resources)
  File "d:\python27\lib\site-packages\nbconvert\nbconvertapp.py", line 335, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "d:\python27\lib\site-packages\nbconvert\exporters\exporter.py", line 165, in from_filename
    return self.from_notebook_node(nbformat.read(f, as_version=4), resources=resources, **kw)
  File "d:\python27\lib\site-packages\nbconvert\exporters\pdf.py", line 145, in from_notebook_node
    rc = self.run_latex(tex_file)
  File "d:\python27\lib\site-packages\nbconvert\exporters\pdf.py", line 108, in run_latex
    self.latex_count, log_error)
  File "d:\python27\lib\site-packages\nbconvert\exporters\pdf.py", line 88, in run_command
    p = subprocess.Popen(command, stdout=stdout, stdin=null, shell=shell, env=env)
  File "d:\python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "d:\python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
```

So I found this piece of code in the pdf.py, it seems that `os.pathsep.join` creates a unicode object while Python 2.x requires it to be a string object. This error would not pop up since in Python 3.x, the string objects are internally implemented as unicode objects
```
env['TEXINPUTS'] = os.pathsep.join([
            cast_bytes_py2(self.texinputs),
            env.get('TEXINPUTS', ''),
        ])
```
I can solve the error by change` env['TEXINPUTS'] ` into a string object by using `.encode('UTF-8')`

**Comment by minrk** [comment]
This has been fixed, we are just working on getting the release (4.2) ready that will have the fix.

**Comment by takluyver** [comment]
Linking the fix for reference: #165
